src/libs/memoize/cache/ArrayCache.ts

```diff
@@ -0,0 +1,82 @@
+import type {Cache, CacheConfig} from './types';
+
+/**
+ * Builder of the cache using `Array` primitive under the hood. It is an LRU cache, where the most recently accessed elements are at the end of the array, and the least recently accessed elements are at the front.
+ * @param config - Cache configuration, check `CacheConfig` type for more details.
+ * @returns
+ */
+function ArrayCache<K, V>(config: CacheConfig<K>): Cache<K, V> {
```

Comment:
Quick question, how do we decide if we build the cache using `Array` or a `Map`?

Reply:
For now, we only have `Array` mode implemented. For more information please check [this discussion](https://callstack-hq.slack.com/archives/C01GTK53T8Q/p1716371827915779).